plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/model/ResolvedConfiguration.kt

```diff
@@ -0,0 +1,11 @@
+package org.gradle.github.dependency.extractor.internal.model
+
+data class ResolvedConfiguration(val rootId: String, val identityPath: String, val name: String, val components: MutableList<ResolvedComponent> = mutableListOf()) {
```

Comment:
I think my original plan with the `ConfigurationDependencyCollector` was to attempt to collect the data into a mutable object, then create an immutable object from it at the end. That way data classes were kept as immutable as possible.

Not saying you need to keep that decision, but I wanted to provide the context for what I was doing here.